[docs] [ENG-11843] Update version source docs

radoslawkrzemien opened the pull request:
# Why

https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote
Companion to: https://github.com/expo/eas-cli/pull/2411

# How

Updated the docs for managing the app versions, after REMOTE has become the recommended option

# Test Plan

Tested manually

## `/app-versions` page
<img width="895" alt="Screenshot 2024-08-06 at 12 06 15" src="https://github.com/user-attachments/assets/5394682d-e5a1-4eaa-9984-5636005b5758">
<img width="895" alt="Screenshot 2024-08-06 at 12 05 51" src="https://github.com/user-attachments/assets/ec3bf0ac-3cb8-455d-a87b-24251118b05a">
<img width="895" alt="Screenshot 2024-08-06 at 12 06 27" src="https://github.com/user-attachments/assets/1752b60c-4c34-465b-aef6-44a435394a75">

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

radoslawkrzemien committed:
[docs] Update version source docs

Updated the docs for managing the app versions, after REMOTE has become the default option

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote

radoslawkrzemien committed:
[docs] Amend versions

Updated the eas-cli versions mentioned in which the version source default is introduced, after the change has been re-classified from chore to breaking change

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote

amandeepmittal commented on docs/pages/build-reference/app-versions.mdx:
```diff
@@ -17,7 +17,9 @@ In managed projects, we use fields `version`/`android.versionCode`/`ios.buildNum
 One of the most frequent causes of app store rejections is submitting a build with a duplicate version number. This happens when a developer forgets to increment the version number before running a build.
 
-EAS Build can manage automatically incrementing these versions for you if you opt into using the "remote" app version source. The default behavior is to use a "local" app version source, which means you control versions manually in their respective config files.
+EAS Build can manage automatically incrementing these versions for you if you use the "remote" app version source, which is the default behaviour. Optionally, you can choose to use a "local" app version source, which means you control versions manually in their respective config files.
```
```suggestion
EAS Build can manage automatically incrementing these versions for you if you use the `remote` app version source, which is the default behaviour. Optionally, you can choose to use a `local` app version source, which means you control versions manually in their respective config files.
```

Remote and local are both properties in **eas.json** so let's use inline syntax highlight here (Unrelated to the changes in the PR).

amandeepmittal commented on docs/pages/build-reference/app-versions.mdx:
```diff
@@ -17,7 +17,9 @@ In managed projects, we use fields `version`/`android.versionCode`/`ios.buildNum
-EAS Build can manage automatically incrementing these versions for you if you opt into using the "remote" app version source. The default behavior is to use a "local" app version source, which means you control versions manually in their respective config files.
+EAS Build can manage automatically incrementing these versions for you if you use the "remote" app version source, which is the default behaviour. Optionally, you can choose to use a "local" app version source, which means you control versions manually in their respective config files.
+
+> **warning** The "remote" version source as a default behaviour has been introduced in `eas-cli` version `10.0.0`. If you are using an older version of CLI, be aware that **"local" will be the default behaviour**.
```
```suggestion
> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `10.0.0`. If you are using an older version of CLI, be aware that **`local` will be the default behavior**.
```

Let's use American spelling as per our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#voice-and-tone).

amandeepmittal reviewed (approved):
Lgtm 👍 Thanks for updating this doc!

I've made some suggestions to improve formatting and how we represent eas.json properties, and some other minor fixes.

radoslawkrzemien committed:
Update docs/pages/build-reference/app-versions.mdx

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

radoslawkrzemien committed:
Update docs/pages/build-reference/app-versions.mdx

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

radoslawkrzemien committed:
Update docs/pages/build-reference/app-versions.mdx

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

radoslawkrzemien committed:
Update docs/pages/build-reference/app-versions.mdx

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

radoslawkrzemien committed:
Update docs/pages/build-reference/app-versions.mdx

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

radoslawkrzemien committed:
[docs] Update docs

Updated the docs to be aligned with the current state of the change and to include the review suggestions

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote

radoslawkrzemien committed:
[docs] Update cli version

There has been a new major version of cli released in the meantime, therefore the correct newest version is mentioned now

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote

radoslawkrzemien committed:
Merge remote-tracking branch 'origin/main' into @radoslawkrzemien/ENG-11843-docs-update-docs-about-default-appversionsource

# Conflicts:
#	docs/pages/build-reference/app-versions.mdx

radoslawkrzemien committed:
[docs] Update docs

Updated the `/app-versions` page

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote

radoslawkrzemien committed:
[docs] Update docs

Updated the `/manage-app-versions` tutorial page

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote

radoslawkrzemien committed:
[docs] Update docs

Don't mention `remote` being enabled by default

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote

amandeepmittal commented on docs/pages/tutorial/eas/manage-app-versions.mdx:
```diff
@@ -89,6 +68,57 @@ After these steps, the app versions will be synced to EAS Build remotely. You ca
 
 </Collapsible>
 
+## Manually manage app version
```
This page is structured in a way where we walk the user through setting up the developer-facing build version values manually and then compare that to how to manage them with EAS automatically. It follows a linear flow. Prior to the updates to the App version management reference page in EAS Build docs, it wasn't clear how to set these values manually or to explain what advantage remote version management provides. That's the context behind why we decided to add this section to the EAS tutorial.

Moving this section below the app version remote causes confusion from a user's perspective about when I've enabled the remote version, why do I need to set these values manually. I think we should avoid doing this right now in this PR since it changes the content and introduces enabling local version support. A proposal I've for upcoming changes (which are out of the scope of this PR):
- Update this section to only mention that manually managing the app version is possible but can be cumbersome.
- Link to the App version management reference page for explanation instead of explaining these concepts here.

Also, the tutorial assumes that the new user persona is installing the EAS CLI and using the latest CLI version. I've created a linear task to follow up on this: https://linear.app/expo/issue/ENG-13013/update-app-version-management-eas-tutorial-chapter-to-account-for after the EAS CLI v11 is released. But for now I don't think we should introduce local version source and the changes below in the tutorial chapter.

radoslawkrzemien replied:
Sounds reasonable, I'll revert changes to this page

amandeepmittal commented on docs/pages/build-reference/app-versions.mdx:
```diff
@@ -34,11 +34,13 @@ When you are doing a production release, the user-facing version should be expli
 ## Remote version source
 
-EAS servers can store and manage your app's developer-facing build version (`android.versionCode` and `ios.buildNumber`) remotely. You can achieve this by setting `cli.appVersionSource` to `remote` in **eas.json**. Then, under the `production` build profile, you can set the `autoIncrement` property to `true`.
+> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, be aware that **`local` will be the default behavior**.
```
```suggestion
> **warning** The `remote` version source is the default behavior from EAS CLI version `11.0.0`. If you are using an older version of the EAS CLI, `local` is the default.
```

- Let's revise the callout to be concise to convey the same information.

amandeepmittal commented on docs/pages/build-reference/app-versions.mdx:
```diff
@@ -34,11 +34,13 @@ When you are doing a production release, the user-facing version should be expli
-EAS servers can store and manage your app's developer-facing build version (`android.versionCode` and `ios.buildNumber`) remotely. You can achieve this by setting `cli.appVersionSource` to `remote` in **eas.json**. Then, under the `production` build profile, you can set the `autoIncrement` property to `true`.
+> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, be aware that **`local` will be the default behavior**.
+
+EAS servers can store and manage your app's developer-facing build version (`android.versionCode` and `ios.buildNumber`) remotely. This is the default behavior and to enable it you need to set `cli.appVersionSource` to `remote` in **eas.json**. Then, under the `production` build profile, you can set the `autoIncrement` property to `true`.
```
```suggestion
EAS servers can store and manage your app's developer-facing build version (`android.versionCode` and `ios.buildNumber`) remotely by default. To enable it, you need to set `cli.appVersionSource` to `remote` in **eas.json**. Then, under the `production` build profile, you can set the `autoIncrement` property to `true`.
```

Let's also mention the default behavior in the first sentence to be concise.

amandeepmittal commented on docs/pages/build-reference/app-versions.mdx:
```diff
@@ -108,11 +110,15 @@ To build your project locally in Android Studio or Xcode using the same version
 ## Local version source
 
-By default, the source of truth for project versions is the local project source code itself. In this case, EAS does not write to the project. It reads the app version values and builds projects as they are. You can opt in to auto incrementing versions locally by setting the `autoIncrement` option on a build profile.
+> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, you **don't need to specify the version source explicitly to `local`**.
```
```suggestion
> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, you **don't need to specify the version source explicitly to `local`**.
```

Let's be concise here too based on the previous comment.

amandeepmittal commented on docs/pages/build-reference/app-versions.mdx:
```diff
@@ -108,11 +110,15 @@ To build your project locally in Android Studio or Xcode using the same version
-By default, the source of truth for project versions is the local project source code itself. In this case, EAS does not write to the project. It reads the app version values and builds projects as they are. You can opt in to auto incrementing versions locally by setting the `autoIncrement` option on a build profile.
+> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, you **don't need to specify the version source explicitly to `local`**.
+
+You can also configure your project so that the source of truth for project versions is the local project source code itself. Add `{ "cli": { "appVersionSource": "local" } }` in your **eas.json**. In this case, EAS does not write to the project. It reads the app version values and builds projects as they are. You can opt in to auto incrementing versions locally by setting the `autoIncrement` opti ...
```
```suggestion
You can configure your project so that the source of truth for project versions is the local project source code itself. To do this, set `cli.appVersionSource` to `local` in your **eas.json**.

With this setup, EAS reads app version values and builds projects as they are. It doesn't write to the project. You can also enable auto incrementing versions locally by setting the `autoIncrement` option on a build profile.
```

- Let's split this into two paragraphs.
- Let's simplify what we have to say about setting `appVersionSource` to `local`. The code block provides the JSON syntax.

amandeepmittal commented on docs/pages/tutorial/eas/manage-app-versions.mdx:
```diff
@@ -19,32 +19,11 @@ An app version is composed of two values:
 ## Automate app version management with EAS Build
 
-EAS Build can assist us in automating these values by utilizing the [remote version resource](/build-reference/app-versions/#remote-version-source). This feature automatically increments developer-facing values and keeps track of them whenever a new production release is made.
+> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, be aware that **`local` will be the default behavior**.
```
```suggestion
```

Let's remove this warning since the first chapter of the tutorial asks the user to install the latest version of EAS CLI.

radoslawkrzemien committed:
[docs] Revert changes to manage-app-versions

Reverted the changes after review comment

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote

radoslawkrzemien committed:
Update docs/pages/build-reference/app-versions.mdx

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

radoslawkrzemien committed:
Update docs/pages/build-reference/app-versions.mdx

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

radoslawkrzemien committed:
Update docs/pages/build-reference/app-versions.mdx

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

amandeepmittal commented on docs/pages/tutorial/eas/manage-app-versions.mdx:
````diff
@@ -19,11 +19,32 @@ An app version is composed of two values:
+```
 
-> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, be aware that **`local` will be the default behavior**.
+After manually adding these values, we need to update them every time before creating a new production build otherwise we will run into a failed submission error. However, to eliminate the chances of running into the error, we can automate this process using EAS Build in the next section
````
```suggestion
After manually adding these values, we need to update them every time before creating a new production build. Otherwise, we will run into a failed submission error. To eliminate the chances of running into this error, we can automate this process using EAS Build in the next section.
```

amandeepmittal reviewed (approved):
Thank you for updating this doc!

szdziedzic commented on docs/pages/build-reference/app-versions.mdx:
```diff
@@ -34,11 +34,13 @@ When you are doing a production release, the user-facing version should be expli
 For developer-facing build version, you can set them to autoincrement on every build. This will help you avoid making manual changes to the project every time you upload a new archive on Play Store testing channels or TestFlight. One common cause for app store rejections is submitting a build with a duplicate version number. It happens when a developer forgets to increment the developer-facing bu ...
 
-EAS Build can help manage developer-facing build versions automatically by incrementing these versions for you if you opt into using the [`remote` version source](#remote-version-source). The default behavior is to use a `local` version source, which means you control versions manually in their respective config files.
+EAS Build can help manage developer-facing build versions automatically by incrementing these versions for you if you opt into using the [`remote` version source](#remote-version-source), which is the default behaviour. Optionally, you can choose to use a `local` app version source, which means you control versions manually in their respective config files.
```
I believe that after multiple review rounds we decided that there is no default value (expect for nonInteractive), but the value needs to be specified explicitly

radoslawkrzemien replied:
Yes, you're right, the docs change reflect the previous state from before the last reviews

radoslawkrzemien committed:
Merge remote-tracking branch 'origin/main' into @radoslawkrzemien/ENG-11843-docs-update-docs-about-default-appversionsource

radoslawkrzemien committed:
[docs] Update version

Updated the version number of cli including the changes mentioned after a new major version had been released in the meantime

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote

radoslawkrzemien committed:
[docs] Update docs

Removed mentions of `remote` being the default and instead the docs say it is recommended

See: https://linear.app/expo/issue/ENG-11843/change-default-appversionsource-to-remote
